editors/sc-ide/core/settings/theme.cpp

```diff
@@ -303,7 +303,7 @@ void Theme::save()
     while (i != mFormats.end()) {
         QTextCharFormat fm = *i.value();
         mSettings->setValue(i.key(), QVariant::fromValue<QTextCharFormat>(fm));
-        i++;
+        ++i;
```

Comment:
This might work (but it doesn't feel particularly elegant)

    for (auto fmt = mFormats.cbegin(); fmt != mFormats.cend(); ++fmt)
    {
        const QTextCharFormat *fm = fmt.value();
        mSettings->setValue(fmt.key(), QVariant::fromValue<QTextCharFormat>(*fm));
    }

Reply:
yeah. i think it's fine to leave it now.